clients/src/test/java/org/apache/kafka/clients/producer/internals/TransactionManagerTest.java

```diff
@@ -2213,6 +2213,27 @@ public void testTransitionToFatalErrorWhenRetriedBatchIsExpired() throws Interru
         assertFalse(transactionManager.hasOngoingTransaction());
     }
 
+    @Test
+    public void testShouldResetProducerStateAfterResolvingSequences() throws InterruptedException, ExecutionException {
+        // Create a TransactionManager without a transactionalId to test
+        // shouldResetProducerStateAfterResolvingSequences.
+        TransactionManager manager = new TransactionManager(logContext, null, transactionTimeoutMs,
+            DEFAULT_RETRY_BACKOFF_MS);
+        assertFalse(manager.shouldResetProducerStateAfterResolvingSequences());
+        TopicPartition tp = new TopicPartition("foo", 0);
```

Comment:
I think the gap in the current tests is that we are only testing with one partition. Indeed this test case also passes without the fix. To trigger the failure, we need one more partition in the test case.

Reply:
That's a fair point.  I revised this so that we now have two partitions. I confirmed that with the revised test, the CME appears when the fix is not present.  (Of course, since the CME is part of "undefined behavior", the CME is not guaranteed in all versions of java, everywhere.)